File: BackEnd/src/domain/quiz.ts
import {
  QuizId,
  QuizOptionId,
  QuizQuestionId,
  SupportProgramId,
  SupportProgramMilestoneId,
  SubjectId,
  UserId,
} from './ids';

export class QuizOption {
  quizOptionId: QuizOptionId;
  text: string;
  isCorrect: boolean;
}

export enum QuizQuestionType {
  MULTIPLE_CHOICE = 'MULTIPLE_CHOICE',
  ESSAY = 'ESSAY',
}

export const ESSAY_ANSWER_MAX_CHARACTERS = 500;

export class QuizQuestion {
  quizQuestionId: QuizQuestionId;
  prompt: string;
  type: QuizQuestionType;
  options: QuizOption[];
  attachments?: string[];
}

export class Quiz {
  quizId: QuizId;
  teacherId?: UserId;
  subjectId?: SubjectId;
  skillFocus?: string;
  supportProgramId?: SupportProgramId;
  milestoneId?: SupportProgramMilestoneId;
  title: string;
  questions: QuizQuestion[];
  createdAt: Date;
}

File: Frontend/GP/Models/Quiz.cs
using System.ComponentModel.DataAnnotations;

namespace GP.Models
{
    public class Quiz : BaseEntity
    {
        [Key]
        public int QuizId { get; set; }
        public int SupportProgramId { get; set; }
        public SupportProgram SupportProgram { get; set; }
        public int MilestoneId { get; set; }
        public Milestone Milestone { get; set; }   
        public string Title { get; set; }
        public List<QuizQuestion> Questions { get; set; }
        public DateTime CreatedAt { get; set; }
    }
}
